File: particle.java
class Particle{
  
  private float x;
  private float y;
  private float vx;
  private float vy;
  private float ax;
  private float ay;
  
  public Particle(float x, float y){
    this.x = x;
    this.y = y;
    vx = 0;
    vy = 0;
    ax = 0;
    ay = 0;
  }
  
  private void updatePosition(float dt){
    x = x + vx*dt + ax*dt*dt/2;
    y = y + vy*dt + ay*dt*dt/2;
  }
  
  private void updateVelocity(float dt){
    vx = vx + ax*dt/2;
    vy = vy + ay*dt/2;
  }
  
  public void updateMotion(float Fx, float Fy, float dt){
    this.updatePosition(dt);
    this.updateVelocity(dt);
    ax = Fx;
    ay = Fy;
    this.updateVelocity(dt);
  }
  
  public void setV(float vx, float vy){
    this.vx = vx;
    this.vy = vy;
  }
  
  public float getX(){
    return x;
  }
  
  public float getY(){
    return y;
  }
  
  public float getVx(){
    return vx;
  }
  
  public float getVy(){
    return vy;
  }
  
  public String toString(){
    return "Position: (" + x + ", " + y + "), " +
           "Velocity: (" + vx + ", " + vy + "), " +
           "Acceleration: (" + ax + ", " + ay + ")";
  
  }
  
}

File: Particle.js
class Particle {

    #x;
    #y
    #vx;
    #vy;
    #ax;
    #ay;
    #m;

    constructor(x, y, m) {
        this.#x = x;
        this.#y = y;
        this.#vx = random() - 0.5;
        this.#vy = random() - 0.5;
        this.#ax = 0;
        this.#ay = 0;
        this.#m = m;
    }

    updatePosition(dt) {
        this.#x = this.#x + this.#vx * dt + this.#ax * dt * dt / 2;
        this.#y = this.#y + this.#vy * dt + this.#ay * dt * dt / 2;
    }

    updateVelocity(dt) {
        this.#vx = this.#vx + this.#ax * dt / 2;
        this.#vy = this.#vy + this.#ay * dt / 2;
    }

    updateAcceleration(Fx, Fy) {
        this.#ax = Fx / this.#m;
        this.#ay = Fy / this.#m;
    }

    getX() {
        return this.#x;
    }

    getY() {
        return this.#y;
    }

    getVx() {
        return this.#vx;
    }

    getVy() {
        return this.#vy;
    }

    getM() {
        return this.#m;
    }

    toString() {
        return "Position: (" + this.#x + ", " + this.#y + "), " +
            "Velocity: (" + this.#vx + ", " + this.#vy + "), " +
            "Acceleration: (" + this.#ax + ", " + this.#ay + ")";
    }

}